ister_gui.py

```diff
@@ -644,6 +672,264 @@ def validate_condition(self):
+        else:
+            Alert('Error!', 'Unable to repair host os').do_alert()
+
+        # leaving host OS chroot
+        os.chdir(old_root)
+        os.chroot('.')
+        os.close(old_root)
+
+        self._umount_host_disk(target_dir, os_disk, boot_part)
+        self.error = False
+        self._action = 'return'
+
+    def update_installer(self, config):
+        """Run swupd update in installer image and restart if successful"""
+        reboot_needed = True
+        watch = WatchableProcess(['swupd', 'update'])
+        watch.start()
+        current = 0
+        Alert('Updating installer', 'Updating installer image then restarting '
+                                    'the installer...', block=False).do_alert()
+        while not watch.done:
+            if len(watch.output) > current:
+                current = len(watch.output)
+                text = '\n'.join(watch.output[-10:])
+                Alert('Updating installer', text, block=False).do_alert()
+        Alert('Updating installer', '\n'.join(watch.output[-10:])).do_alert()
+
+        if 'System already up-to-date' in '\n'.join(watch.output[-10:]):
+            reboot_needed = False
+
+        if not watch.poll():
+            if reboot_needed:
+                Alert('Updating installer',
+                      'Successful update, rebooting...').do_alert()
+                subprocess.call('reboot')
+            else:
+                Alert('Updating installer',
+                      'No update available, no reboot required').do_alert()
+        else:
+            Alert('Error!', 'Unable to update installer').do_alert()
+
+        self._action = 'return'
+
+    def _find_current_disk(self):
+        """
+        Find the current disk so it can be skipped when searching for a Linux
+        root
+        """
+        cmd = ['lsblk', '-l', '-o', 'NAME,MOUNTPOINT']
+        output = subprocess.check_output(cmd).decode('utf-8')
+        for line in output.split('\n'):
+            if '/' in line:
+                return line.split()[0]
+
+        return ''
+
+    def _mount_host_disk(self, config):
+        """Search for and mount the host os disk"""
+        os_disk = ''
+        prefix = 'ister-{}-'.format(str(config["Version"]))
+        target_dir = tempfile.mkdtemp(prefix=prefix)
+        disks = get_list_of_disks()
+        current = self._find_current_disk()
+        boot_part = ''
+        for disk in disks:
+            part_info = get_disk_info('/dev/{}'.format(disk))
+            for part in part_info['partitions']:
+                if part.get('type') and 'EFI' in part['type']:
+                    boot_part = part['name']
+                if part.get('type')                \
+                        and 'root' in part['type'] \
+                        and current not in part['name']:
+                    os_disk = part['name']
+                if os_disk and boot_part:
+                    break
+            if os_disk and boot_part:
+                break
+
```

Comment:
It may be good to break out the mount try to another function and make a test if os_disk has any data or not before calling that function and having a different error in that case.

Similarly for boot_part aside from it doesn't need to be an error just a notification that we couldn't find a boot partition to mount (vs an error mounting a boot partition we did find).